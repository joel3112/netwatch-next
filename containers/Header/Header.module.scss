/** Container (main component) **/

.wrapper {
  @include box(100%, 100%);

  position: relative;
}

.logo {
  width: var(--size-header);
  height: var(--size-header);
  content: var(--img-logo);
}

.title {
  color: var(--color-text);
  font-family: var(--font-logo);
  margin: 5px 15px 0 10px !important;
}

/** HeaderActions (child component) **/

.headerAction {
  @include box(var(--size-header-action), var(--size-header-action) !important);

  border-color: var(--color-border) !important;
  min-width: var(--size-header-action);
  font-size: 1.4rem !important;
  padding: 0 !important;

  &:hover {
    border-color: var(--color-primary) !important;
  }

  &.search {
    width: var(--size-search-input) !important;
    justify-content: flex-start;
    padding: 0 15px !important;

    span {
      font-size: rem(15px) !important;
      color: var(--color-text);
    }
  }
}

/** HeaderSearch (child component) **/

.modalSearch {
  top: var(--top-search-modal);
  transform: translateX(-50%);
}
