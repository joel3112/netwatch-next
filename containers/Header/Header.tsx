import { useState } from 'react';
import Link from 'next/link';
import Image from 'next/image';
import { IconType } from 'react-icons';
import { FiSearch, FiSettings } from 'react-icons/fi';
import { ElementHTML } from '@/types';
import { useI18n } from '@/hooks/useI18n';
import { useBreakpoint } from '@/hooks/useBreakpoint';
import { Settings } from '@/containers/Settings';
import { MediaAutocomplete } from '@/containers/MediaAutocomplete';
import { Space } from '@/components/layout';
import { Button } from '@/components/forms';
import { Heading } from '@/components/typography';
import { Drawer, Portal, Modal } from '@/components/overlay';
import { classes } from '@/utils/helpers';
import styles from '@/containers/Header/Header.module.scss';

/* -------------------------------------------------------------------------- */
/** HeaderSearch (child component) **/
/* -------------------------------------------------------------------------- */

const HeaderSearch = () => {
  const { t } = useI18n();
  const { isMobile } = useBreakpoint();
  const [opened, setOpened] = useState(false);

  const buttonProps = (text: boolean) => ({
    onClick: () => setOpened(true),
    outline: true,
    className: classes(styles.headerAction, text && styles.search)
  });

  const handleChange = () => setOpened((prev) => !prev);

  return (
    <Space align="center" gap={15} className={styles.headerActions}>
      {!isMobile ? (
        <Button {...buttonProps(true)}>
          <FiSearch />
          <span>{t('header.action.search')}</span>
        </Button>
      ) : (
        <Button tooltip={t('header.action.search')} {...buttonProps(false)}>
          <FiSearch />
        </Button>
      )}
      <Portal.Paper>
        <Modal opened={opened} onChange={handleChange} className={styles.modalSearch}>
          <MediaAutocomplete onClose={() => setOpened(false)} />
        </Modal>
      </Portal.Paper>
    </Space>
  );
};

/* -------------------------------------------------------------------------- */
/** HeaderActions (child component) **/
/* -------------------------------------------------------------------------- */

type HeaderActionsProps = {
  actions?: Array<{ label: string; Icon: IconType }>;
};

const defaultActions = [{ label: 'header.action.settings', Icon: FiSettings }];

const HeaderActions = ({ actions = defaultActions }: HeaderActionsProps) => {
  const { t } = useI18n();

  return (
    <Space align="center" gap={20} className={styles.headerActions}>
      {actions.map(({ label, Icon }) => (
        <Portal key={label}>
          <Portal.Handler>
            <Button outline tooltip={t(label)} className={styles.headerAction}>
              <Icon />
            </Button>
          </Portal.Handler>
          <Portal.Paper>
            <Drawer position="right" heading={t(label)}>
              <Settings />
            </Drawer>
          </Portal.Paper>
        </Portal>
      ))}
    </Space>
  );
};

/* -------------------------------------------------------------------------- */
/** Container (main component) **/
/* -------------------------------------------------------------------------- */

export type HeaderProps = typeof defaultProps & ElementHTML;

const defaultProps = {};

const Header = ({}: HeaderProps) => {
  const { t } = useI18n();

  return (
    <Space justify="between" align="center" className={classes(styles.wrapper)}>
      <Link href="/">
        <a>
          <Space align="center">
            <div className={styles.logo}>
              <Image src="/assets/images/logo-light.png" alt="logo" layout="fill" />
            </div>

            <Heading level={1} className={styles.title}>
              {t('application.name')}
            </Heading>
          </Space>
        </a>
      </Link>

      <Space align="center" gap={10}>
        <HeaderSearch />
        <HeaderActions />
      </Space>
    </Space>
  );
};

Header.defaultProps = defaultProps;

export default Header;
