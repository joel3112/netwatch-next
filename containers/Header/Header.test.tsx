import { render, screen } from '@testing-library/react';
import Header from '@/containers/Header/Header';

describe('Tests Header component', () => {
  test('renders container correctly', () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });

  test('renders title correctly', () => {
    render(<Header />);

    expect(screen.getByRole('heading')).toBeInTheDocument();
  });

  test('renders logo image url correctly', () => {
    render(<Header />);

    expect(screen.getByAltText('logo')).toBeInTheDocument();
  });
});
