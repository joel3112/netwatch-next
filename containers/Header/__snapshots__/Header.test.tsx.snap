// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Header component renders container correctly 1`] = `
<div>
  <div
    class="wrapper wrapper"
  >
    <a
      href="/"
    >
      <div
        class="wrapper"
      >
        <div
          class="logo"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="logo"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
        <h1
          class="wrapper title"
        >
          application.name
        </h1>
        <style>
          
        .wrapper {
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          
          
          align-items: center;
        }
      
        </style>
      </div>
    </a>
    <div
      class="wrapper"
    >
      <div
        class="wrapper headerActions"
      >
        <button
          class="wrapper medium outline headerAction search"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <line
              x1="21"
              x2="16.65"
              y1="21"
              y2="16.65"
            />
          </svg>
          <span>
            header.action.search
          </span>
        </button>
        <style>
          
        .wrapper {
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          gap: 15px;
          
          align-items: center;
        }
      
        </style>
      </div>
      <div
        class="wrapper headerActions"
      >
        <div
          class="wrapper"
        >
          <div
            aria-label="header.action.settings"
            class="content"
            data-mui-internal-clone-element="true"
          >
            <button
              aria-label="header.action.settings"
              class="wrapper medium outline headerAction"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="3"
                />
                <path
                  d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
                />
              </svg>
            </button>
          </div>
        </div>
        <style>
          
        .wrapper {
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          gap: 20px;
          
          align-items: center;
        }
      
        </style>
      </div>
      <style>
        
        .wrapper {
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          gap: 10px;
          
          align-items: center;
        }
      
      </style>
    </div>
    <style>
      
        .wrapper {
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          
          justify-content: space-between;
          align-items: center;
        }
      
    </style>
  </div>
</div>
`;
